doc/source/dsintro.rst

```diff
@@ -243,12 +260,21 @@ not matching up to the passed index.
 If axis labels are not passed, they will be constructed from the input data
 based on common sense rules.
 
+.. note::
+
+   When the data is a dict, and columns is not passed, the DataFrame columns
```

Comment:
"and columns is not passed" -> I would try to make it clear that it is about a keyword `columns`, by either qouting it, or "no columns are passed", or "`columns` is not specified"